frontend/openchat-client/src/openchat.ts

```diff
@@ -1056,16 +1057,44 @@ export class OpenChat extends EventTarget {
             .then((resp) => {
                 const success = resp === "success";
                 if (!success) {
-                    undelete();
+                    _undelete();
                 }
                 return success;
             })
             .catch((_) => {
-                undelete();
+                _undelete();
                 return false;
             });
     }
 
+    undeleteMessage(
+        chatId: string,
+        threadRootMessageIndex: number | undefined,
+        messageId: bigint
+    ): Promise<boolean> {
+        const chat = this._liveState.chatSummaries[chatId];
+
+        if (chat === undefined) {
+            return Promise.resolve(false);
+        }
+
+        undeletingMessagesStore.add(messageId);
+
+        return this.api
+            .undeleteMessage(chat.kind, chatId, messageId, threadRootMessageIndex)
+            .then((resp) => { 
+                const success = resp.kind === "success";
+                if (success) {
+                    localMessageUpdates.markUndeleted(messageId.toString(), resp.message.content);
+                }
+                return success;
+            })
+            .catch((_) => false)
```

Comment:
I think in other cases we would typically log the exact failure response here - otherwise we won't really know exactly what went wrong.

Reply:
Yes ok - I used deleteMessage as a template...